packages/statemanager/src/rpcStateManager.ts

```diff
@@ -16,26 +16,24 @@ import {
 import debugDefault from 'debug'
 import { keccak256 } from 'ethereum-cryptography/keccak.js'
 
-import { AccountCache, OriginalStorageCache, StorageCache } from './cache/index.js'
-import * as Capabilities from './capabilities.js'
+import { AccountCache, CacheType, OriginalStorageCache, StorageCache } from './cache/index.js'
+import { modifyAccountFields } from './util.js'
 
-import type { RPCStateManagerOpts } from './index.js'
-import type { AccountFields, Proof, StateManagerInterface, StorageDump } from '@ethereumjs/common'
+import type { Proof, RPCStateManagerOpts } from './index.js'
+import type { AccountFields, StateManagerInterface, StorageDump } from '@ethereumjs/common'
 import type { Address, PrefixedHexString } from '@ethereumjs/util'
 import type { Debugger } from 'debug'
 
 const KECCAK256_RLP_EMPTY_ACCOUNT = RLP.encode(new Account().serialize()).slice(2)
 
 export class RPCStateManager implements StateManagerInterface {
   protected _provider: string
+  protected _contractCache: Map<string, Uint8Array>
+  protected _storageCache: StorageCache
   protected _blockTag: string
-
-  _accountCache: AccountCache
-  _storageCache: StorageCache
-  _contractCache: Map<string, Uint8Array>
-
+  protected _accountCache: AccountCache
```

Comment:
The caches in `RPCStateManager` will still be transfered to the new interface, right?

Reply:
The `contractCache` seems different in the rpcStateManager than in the others (i.e. not the `ContractCache` class). It would indeed be nice to refactor in a similar fashion here. @acolytec3, why do we use a custom Map<string, Uint8Array> here rather than the `ContractCache` class?

Reply:
I am pretty sure that just some of the inconsistencies we have in state manager and that was rather to have some simple version in and so you can just fully and 1:1 with the original version replace.

Reply:
I'm 99% sure that is just because the original implementation from the RPCstatemanager just didn't have caches at all. No compelling reason though and the cache will do the same thing.

Reply:
Just update to use the same _caches property and type as the other 2